### DISCUSSION TITLE
Northstar LLM

### DISCUSSION THREAD

**Discussion by GeckoEidechse** [R2Northstar/Northstar#588] [Ideas]
(Taken from this Discord discussion https://discord.com/channels/920776187884732556/1104763255332810923/1104763327151869982)

We doing AI now let's gooooo!!111!1!one!1!eleven!!11!

No but for real jokes aside, as much as LLMs are just stochastic parrots, having a stochastic parrot that helps generate code or answers tickets when no helper is around would be kinda cool. So let's see where this would be useful:

### Code generation:

Northstar was published after the GPT3.5 and GPT4 cutoff dates for its training data. As such it has no clue about Northstar's existence entirely, including code snippets from it.

Further Respawn uses a custom flavour of Squirrel for Titanfall2 so we cannot use ChatGPT to generate any code snippets for Northstar.

So what if we trained a fork of an existing LLM on Titanfall2/Northstar code snippets to allow for generating code.
Of course a limiting factor is that the amount of training data that would exist for it is tiny, even when taking mods published to Thunderstore into account.

### Tickets:

The majority of tickets tend to be them same kind of issues revolving around broken mods, failed manual updates, issues with port-forwarding, or EA app shenanigans.
Even though those issues are covered in our wiki, we still often get tickets for those issues.

So what if we could feed the wiki into an LLM to add it to it's knowledge base and have it aid with basic troubleshooting. Of course, precautions need to be made so that it doesn't tell users to "delete the French language pack" (i.e. `sudo rm -fr /*`).

### Other:

idk? Your ideas here :>

### Useful links

This is a fast moving field so these links might already be outdated again soon.

- https://github.com/imartinez/privateGPT
- https://github.com/openchatai/OpenChat
- https://github.com/Mintplex-Labs/anything-llm
- https://github.com/n4ze3m/dialoqbase
- https://ai.meta.com/llama/
- https://sh.itjust.works/post/1555047
- https://hacks.mozilla.org/2023/07/so-you-want-to-build-your-own-open-source-chatbot/
- Ollama: https://github.com/jmorganca/ollama
- A GPT (holy fuck why do OpenAI call it that it sounds so stupid) with Northstar wiki ingested: https://chat.openai.com/g/g-e9YjxAeiK-northstar-wiki

**Comment by ASillyNeko** [comment]
When is this coming out?
